troubleshooting/index.md

```diff
@@ -1,44 +1,620 @@
+figure out exactly where the error is occurring. For more context when you're
+viewing the results of the commands directly in the terminal, you may want to
+pass the verbose flag.
+
+#### Debug via Tugboat's CLI
+
+If you've installed [Tugboat's CLI](../tugboat-cli/index.md), you can debug your
+Preview build from the comfort of your own terminal. Type `tugboat help` for a
+list of commands and options. Following our debug process above, you might:
+
+```shell
+tugboat log 5b04c7d14c3dad00016a2e80
+tugboat ls services preview=5b04c7d14c3dad00016a2e80
+tugboat shell 5d26155195433d189bdf9307
+```
+
+And once you've got a shell into the Service, you can go line-by-line debugging
+your config file, just the same as if you'd used the web UI to terminal into a
+Service.
+
+#### Debug via pushing changes to git
+
+If you think you know where the problem is in your config file, you can always
+make the changes, make a PR or push a branch to your git repo, and try again to
+[build the Preview](../building-a-preview/index.md#how-to-build-a-preview).
+
+The same caveats apply as in
+[Debugging changes to your Tugboat config file](#debugging-changes-to-your-tugboat-config-file)
+above; if you're making changes when a Tugboat project has a Base Preview,
+you'll need to manually rebuild a Preview from scratch if you're using a Base
+Preview in order to apply config file changes. For more info, see:
+[The build process: explained](../building-a-preview/index.md#the-build-process-explained).
+
+## How to check the Preview logs
+
+You can click into the Preview's name to view the Preview's logs _and_ logs for
+individual Services you're running on a Preview. Alternately, you could use the
+[Tugboat CLI](../tugboat-cli/index.md) to view
+[Preview logs](../tugboat-cli/index.md#view-preview-logs) and
+[Service logs](../tugboat-cli/index.md#view-services-logs).
+
+![View Preview logs](_images/click-into-preview-for-logs.png)
+
+### How to fix problem X in config file
+
+- [PHP out of memory issues](#php-out-of-memory-issues)
+- [MySQL server has gone away](#mysql-server-has-gone-away)
+- [cd isn't working](#cd-isnt-working)
+- [Tugboat error messages](#tugboat-error-messages)
+- [Running a background process "breaks" the build](#running-a-background-process)
+
+#### PHP out of memory issues
+
+If you're getting "PHP out of memory errors", you can manually set the memory
+limit higher. If you're using one of the
+[tugboatqa php images](../setting-up-services/index.md#tugboats-prebuilt-docker-images),
+use something like this in your build script:
+
+`echo "memory_limit = 512M" >> /usr/local/etc/php/conf.d/my-php.ini`
+
+Or you might try something like this in your drupal settings.php:
+
+`if (drupal_is_cli()) { ini_set('memory_limit', '-1');`
+
+#### MySQL server has gone away
+
+If you're getting a "MySQL server has gone away" error, you can resolve this by
+increasing the packet size for MySQL:
+
+_BEN CAN WE INSERT AN EXAMPLE FOR THIS?_
```

Comment:
@q0rban Can we get a snippet of the command to run when the "MySQL server has gone away"?